staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -224,6 +228,7 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 			glog.V(4).Infof("decoding stream as YAML")
 			d.decoder = NewYAMLToJSONDecoder(buffer)
 		}
+		d.decoder.DisallowUnknownFields()
```

Comment:
this is not a backwards-compatible change. needs to be conditional

Reply:
Conditional on what?

Reply:
whether the constructor of the YAMLOrJSONDecoder wanted strict mode, e.g. an option in YAMLOrJSONDecoder.

also, no need to make DisallowUnknownFields() part of the decoder interface, especially when it is a no-op for the yaml decoder, just call it on the json.NewDecoder in the `isJSON` block